dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc

```diff
@@ -705,41 +709,52 @@ class updatedns
                 curl_setopt($ch, CURLOPT_URL, $server);
                 break;
             case 'cloudflare':
-                $dnsServer ='api.cloudflare.com';
-                $dnsHost = str_replace(' ', '', $this->_dnsHost);
-                $host_names = explode('.', $dnsHost);
-                $bottom_host_name = $host_names[count($host_names) - 2] . '.' . $host_names[count($host_names) - 1];
+            case 'cloudflare-v6':
+                $baseUrl = 'https://api.cloudflare.com/client/v4';
+                $fqdn = str_replace(' ', '', $this->_dnsHost);
+                $fqdnParts = explode('.', $fqdn);
+                $hostName = array_shift($fqdnParts);
+                $domainName = implode('.', $fqdnParts);
+                $recordType = ($this->_useIPv6) ? 'AAAA' : 'A';
+                $hostData = array(
+                    "content" => "{$this->_dnsIP}",
+                    "type" => $recordType,
+                    "name" => $fqdn
+                );
 
                 curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
                 curl_setopt($ch, CURLOPT_HTTPHEADER, array(
-                    'X-Auth-Email: '.$this->_dnsUser.'',
-                    'X-Auth-Key: '.$this->_dnsPass.'',
+                    "X-Auth-Email: $this->_dnsUser",
```

Comment:
it probably doesn't matter functionally, but for consistency we try to write i.e. {$this->_dnsUser} and {$this->_dnsPass} in double-quoted strings

Reply:
Updated the PR for these